python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/launcher_tests/test_container_context.py

```diff
@@ -70,3 +97,20 @@ def test_merge(
         empty_container_context.merge(secrets_container_context).secrets_tags
         == secrets_container_context.secrets_tags
     )
+
+    merged = merged.merge(environment_container_context)
+
+    assert (
+        merged.environment
+        == environment_container_context.environment + secrets_container_context.environment
+    )
+
+    merged = merged.merge(other_environment_container_context)
+
+    assert len(merged.environment) == 4
+    assert merged.environment == [
+        {"name": "Hello", "value": "Goodbye"},
+        {"name": "FOO", "value": "BAZ"},
+        {"name": "FOO", "value": "BAR"},
```

Comment:
Yeah, I agree. This is not what I would expect when merging. I don't really know if ECS will resolve that for us or if it'll error 🤷